core/templates/components/forms/schema-based-editors/schema-based-float-editor.component.spec.ts

```diff
@@ -0,0 +1,247 @@
+
+  it('should call input focus when user selects the input field', () => {
+    spyOn(component.inputFocus, 'emit');
+
+    component.onFocus();
+
+    expect(component.inputFocus.emit).toHaveBeenCalled();
+  });
+
+  it('should call input blur user deselects the input field', () => {
+    spyOn(component.inputBlur, 'emit');
+
+    component.onBlur();
+
+    expect(component.inputBlur.emit).toHaveBeenCalled();
+  });
+
+  it('should get minimum and maximum values', () => {
+    expect(component.getMinValue()).toBe(1.1);
+    expect(component.getMaxValue()).toBe(3.5);
+  });
+
+  it('should not register keyboard event when user is typing', fakeAsync(() => {
+    let evt = new KeyboardEvent('', {
+      keyCode: 14
+    });
+
+    component.userIsCurrentlyTyping = false;
+
+    component.onKeypress(evt);
+
+    expect(component.userIsCurrentlyTyping).toBe(true);
+  }));
+
+  it('should register keyboard event when user is not typing if there' +
+    'was as an error in form then set userTyping to false', fakeAsync(() => {
+    let evt = new KeyboardEvent('', {
+      keyCode: 13
+    });
+
+    let formvalue = new FormControl(null);
+    formvalue.setErrors({invalidNumber: true});
+    component.floatForm.form.controls.floatValue = formvalue;
+    component.userIsCurrentlyTyping = true;
+
+    component.onKeypress(evt);
+
+    expect(component.userIsCurrentlyTyping).toBe(false);
+  }));
+
+  it('should register keyboard event when user is not typing if there' +
+    'was not an error emit value', fakeAsync(() => {
+    spyOn(schemaFormSubmittedService.onSubmittedSchemaBasedForm, 'emit')
+      .and.stub();
+    let evt = new KeyboardEvent('', {
+      keyCode: 13
+    });
+
+    let formvalue = new FormControl(null);
+    formvalue.setErrors({});
+    component.floatForm.form.controls.floatValue = formvalue;
+    component.userIsCurrentlyTyping = true;
+
+    component.onKeypress(evt);
+
+    expect(component.userIsCurrentlyTyping).toBe(true);
+    expect(schemaFormSubmittedService.onSubmittedSchemaBasedForm.emit)
+      .toHaveBeenCalled();
+  }));
+
+  it('should generate error for wrong input', fakeAsync(() => {
+    spyOn(numericInputValidationService, 'validateNumber')
+      .and.returnValue('I18N_INTERACTIONS_NUMERIC_INPUT_INVALID_NUMBER');
+    component.localValue = null;
+    component.generateErrors();
+
+    expect(component.errorStringI18nKey)
+      .toBe('I18N_INTERACTIONS_NUMERIC_INPUT_INVALID_NUMBER');
+  }));
+
+
+  it('should validate value', () => {
+    component.uiConfig = null;
+
+    expect(component.validate(new FormControl(2)))
+      .toEqual({});
+    expect(component.validate(new FormControl(null)))
+      .toEqual({error: 'invalid'});
+  });
+
+  it('should get current decimal separator', ()=>{
+    spyOn(numberConversionService, 'currentDecimalSeparator')
+      .and.returnValues('.', ',');
+
+    expect(component.currentDecimalSeparator()).toEqual('.');
+    expect(component.currentDecimalSeparator()).toEqual(',');
```

Comment:
Kept it as it is in develop. But I think this test makes sense. If you want we can discuss more on why mocking further up the chain won't solve anything for us.

Reply:
Can you change the method name to start with a verb?

Also I just realized the expectation is on the component and the mock is on the service, so that part is ok -- sorry for misunderstanding.

Reply:
Done!